hearth/lib/hearth/event_stream/async_output.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require_relative '../output'
+
+module Hearth
+  module EventStream
+    # Represents an Asynchronous output from an Event Stream operation.
+    class AsyncOutput < Output
```

Comment:
Yeah - I debated this a lot.  I landed on the AsyncOutput should inherit from Output so that the rest of the middleware stack will continue to work.  AsyncOutput's may have `error` set (when there are initial connection errors).  

I don't think these can or should live in `Output` - all of this logic and additional methods don't apply to non-event stream responses.